tools/go.mod

```diff
@@ -1,3 +0,0 @@
-module github.com/pulumi/docs/tools
```

Comment:
@lukehoban looks like you added this module file recently. In this PR, I added a module directly for `resourcedocsgen`, so I removed this one. Is there a preference to keep this over what I have done?